src/normalizer/nonspacing_mark.rs

```diff
@@ -0,0 +1,162 @@
+use std::borrow::Cow;
+use std::collections::HashSet;
+
+use once_cell::sync::Lazy;
+
+use super::{Normalizer, NormalizerOption};
+use crate::detection::{Language, Script};
+use crate::Token;
+
+static HASHSET: Lazy<HashSet<u32>> = Lazy::new(|| {
+    let bytes = include_bytes!("../../dictionaries/bin/nonspacing_mark/marks.bin");
+
+    HashSet::from_iter(
+        bytes.chunks_exact(4).map(|chunk| u32::from_ne_bytes(chunk.try_into().unwrap())),
+    )
+});
+
+/// A global [`Normalizer`] removing nonspacing marks.
+///
+/// This normalizer uses built-in `HashSet` internally to check over the marks set
+pub struct NonspacingMarkNormalizer;
+
+impl Normalizer for NonspacingMarkNormalizer {
+    fn normalize<'o>(
+        &self,
+        mut token: Token<'o>,
+        options: NormalizerOption,
+    ) -> Box<dyn Iterator<Item = Token<'o>> + 'o> {
+        if token.lemma().chars().any(is_nonspacing_mark) {
+            let mut lemma = String::new();
+            let mut char_map = if options.create_char_map { Some(Vec::new()) } else { None };
```

Comment:
```suggestion
            let mut char_map = options.create_char_map.then_some(Vec::new);
```
Sorry, I know I'm reviewing a copy-pasted code, but, since the `1.62.0`we can directly use [then_some](https://doc.rust-lang.org/std/primitive.bool.html#method.then_some) on booleans ✨

Reply:
Thanks, I've also changed that for the Chinese normalizer :)